### ISSUE TITLE
[BUG] Unable to create a cluster network with the Harvester terraform provider

### ISSUE THREAD

**Issue by juadk** [harvester/harvester#1042]
**Describe the bug**
I was trying to create a cluster network following the instructions in this [issue](https://github.com/harvester/harvester/issues/634) but terraform told me the resource already exists whereas my cluster is just out-of-the-box.

```
harvester_clusternetwork.vlan: Creating...

 Error: clusternetworks.network.harvesterhci.io "vlan" already exists
 
   with harvester_clusternetwork.vlan,
   on cluster_network.tf line 1, in resource "harvester_clusternetwork" "vlan":
    1: resource "harvester_clusternetwork" "vlan" {
```

If I'm not wrong, this terraform resource targets the vlan setting just below:
![Screenshot from 2021-07-26 17-02-23](https://user-images.githubusercontent.com/6025636/127011880-98d4137c-9222-4db2-8bcb-499a511aa906.png)

As you can see, nothing is configured.

**To Reproduce**
1. Follow the step to configure the harvester terraform provider [here](https://github.com/harvester/harvester/issues/634#issuecomment-884778622)
2. Follow the step to create a [cluster network](https://github.com/harvester/terraform-provider-harvester/blob/master/examples/resources/harvester_clusternetwork/resource.tf)
3. Apply the configuration with terraform.
` Error: clusternetworks.network.harvesterhci.io "vlan" already exists`

**Expected behavior**
Terraform should be able to create the network.

**Support bundle**
[harvester-supportbundle_58a64e6c-b087-49b5-89d2-6b3196a32355_2021-07-26T15-04-53Z.zip](https://github.com/harvester/harvester/files/6879251/harvester-supportbundle_58a64e6c-b087-49b5-89d2-6b3196a32355_2021-07-26T15-04-53Z.zip)

**Environment:**
 - Harvester ISO version: master-2f810069-head + terraform-provider-harvester v.0.1.0 + Terraform v1.0.3
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 5 nodes cluster DL360 G9

**Additional context**

**Comment by futuretea** [comment]
If a resource is already exists, need to use `terraform import` to import it first

**Comment by juadk** [comment]
> If a resource is already exists, need to use `terraform import` to import it first

That means we will never be able to create a resource for the cluster network because the name is always vlan, right?

![clusternet1](https://user-images.githubusercontent.com/6025636/127123606-fc0d5745-2edc-4bda-a8f0-6d64a7a17afe.png)


I tried to import it:
```
terraform import harvester_clusternetwork.vlan default/vlan
harvester_clusternetwork.vlan: Importing from ID "default/vlan"...
harvester_clusternetwork.vlan: Import prepared!
  Prepared harvester_clusternetwork for import
harvester_clusternetwork.vlan: Refreshing state... [id=default/vlan]
╷
│ Error: Cannot import non-existent remote object
│ 
│ While attempting to import an existing object to "harvester_clusternetwork.vlan", the provider detected that no object exists with
│ the given id. Only pre-existing objects can be imported; check that the id is correct and that it is associated with the
│ provider's configured region or endpoint, or use "terraform apply" to create a new remote object for this resource.
╵
```
I wasn't able to modify it too:
```
terraform apply 

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  + create

Terraform will perform the following actions:

  # harvester_clusternetwork.vlan will be created
  + resource "harvester_clusternetwork" "vlan" {
      + default_physical_nic = "eth1"
      + enable               = true
      + id                   = (known after apply)
      + name                 = "vlan"
      + namespace            = "harvester-system"
      + state                = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

harvester_clusternetwork.vlan: Creating...
╷
│ Error: clusternetworks.network.harvesterhci.io "vlan" already exists
│ 
│   with harvester_clusternetwork.vlan,
│   on cluster_network.tf line 1, in resource "harvester_clusternetwork" "vlan":
│    1: resource "harvester_clusternetwork" "vlan" {
│ 
╵
```
I don't see how to activate the setting and configure it to eth1.

**Comment by futuretea** [comment]
> That means we will never be able to create a resource for the cluster network because the name is always vlan, right?

yes.

```bash
terraform import harvester_clusternetwork.vlan harvester-system/vlan
```

**Comment by juadk** [comment]
My bad, I got tricked by the namespace... I forgot that the settings are under the harvester-system namespace... let me try again, thx.

**Comment by juadk** [comment]
All is ok with the harvester-system namespace, let's close this issue.

**Comment by ldevulder** [comment]
@futuretea I would like to reopen the issue, at least for discussion.

FMPOV there is an issue here, as the `vlan` resource is created by default in Harvester (and that's expected) but in that case it cannot be created by the TF provider unless an import command is done before. After this import I can destroy and then recreate the `vlan` resource without any issue (if I set ` prevent_destroy = false` in `lifecycle` of course), so why the first creation couldn't be done/bypass if the resource already existing? (TF problem here?)
As a customer/sysadmin I thinks that's it's not very convenient to do this way, it happens now for `vlan` but could be the same later when new functionalities will be added to the TF provider. How a customer could know what to import to the first time?

**Comment by ldevulder** [comment]
@yasker WDYT on this?

**Comment by futuretea** [comment]
> why the first creation couldn't be done/bypass if the resource already existing? (TF problem here?)

@ldevulder  Yes, This is terraform's machenism.

**Comment by futuretea** [comment]
> How a customer could know what to import to the first time?

Customers can use terraformer to generate and import existing resources.
https://github.com/harvester/harvester/issues/943#issuecomment-886431707
https://github.com/harvester/terraformer

**Comment by guangbochen** [comment]
> As a customer/sysadmin I thinks that's it's not very convenient to do this way, it happens now for vlan but could be the same later when new functionalities will be added to the TF provider. How a customer could know what to import to the first time?

Currently, we are refactoring the VLAN network setting of the 0.3 release, therefore we can come back and revisit this type of resource later, and generally speaking, those `settings` configuration will only need to be set once by admin only, and normally they won't change it after it is configured, that's why that type resource is not covered in the first release.

**Comment by futuretea** [comment]
clusternetwork has been change to non-namespaced, in terraform-provider-harvester v0.2.0 + harvester v0.3.0, we don't need to specify namespace:

```yaml
resource "harvester_clusternetwork" "vlan" {
  lifecycle {
    prevent_destroy = true
  }
  name                 = "vlan"
  enable               = true
  default_physical_nic = "eth0"
}
```

https://github.com/harvester/terraform-provider-harvester/releases/tag/v0.2.0

https://github.com/harvester/terraform-provider-harvester/blob/master/examples/resources/harvester_clusternetwork/resource.tf

**Comment by yasker** [comment]
~~@slickwarren can you help to validate this?~~

Need to reassign due to Caleb is on other tasks

**Comment by yarunachalam** [comment]
validated cluster network without namespace in the resource. It works as expected

resource "harvester_clusternetwork" "vlan" {
  lifecycle {
    prevent_destroy = true
  }
  name                 = "vlan"
  enable               = true
  default_physical_nic = "eth0"
}

**Comment by chajain** [comment]
Verified by @yarunachalam
